fast_node.go

```diff
@@ -0,0 +1,41 @@
+package iavl
+
+import (
+	"github.com/pkg/errors"
+)
+
+// NOTE: This file favors int64 as opposed to int for size/counts.
+// The Tree on the other hand favors int.  This is intentional.
+
+type FastNode struct {
+	key                  []byte
+	versionLastUpdatedAt int64
+	value                []byte
+	leafHash             []byte // TODO: Look into if this would help with proof stuff.
+}
+
+// NewFastNode returns a new fast node from a value and version.
+func NewFastNode(value []byte, version int64) *FastNode {
+	return &FastNode{
+		versionLastUpdatedAt: version,
+		value:                value,
+	}
+}
+
+// MakeFastNode constructs an *FastNode from an encoded byte slice.
+func MakeFastNode(buf []byte) (*FastNode, error) {
```

Comment:
Should we call this `DeserializeFastNode` (its a naming issue in existing code atm)

Reply:
I think this is a much better naming convention since that is what is actually going on here.